pkg/sql/refresh_materialized_view.go

```diff
@@ -80,6 +77,10 @@ func (n *refreshMaterializedViewNode) startExec(params runParams) error {
 	// results of the view query into the new set of indexes, and then change the
 	// set of indexes over to the new set of indexes atomically.
 
+	if !params.p.extendedEvalCtx.TxnIsSingleStmt {
+		return pgerror.Newf(pgcode.InvalidTransactionState, "cannot refresh view in a multi-statement transaction")
+	}
```

Comment:
can you add a test in materialized_view_test.go

right below https://github.com/cockroachdb/cockroach/blob/991191666217e5cf6347395ec0ff883a2b80c3e6/pkg/sql/materialized_view_test.go#L70

you can add
```
	// Verify that refreshing with a prepared statement works.
	preparedStmt, err := sqlDB.Prepare(`REFRESH MATERIALIZED VIEW t.v;`)
	if err != nil {
		t.Fatal(err)
	}
	if _, err := preparedStmt.Exec(); err != nil {
		t.Fatal(err)
	}
```